### ISSUE TITLE
Verify email in Riot

### ISSUE THREAD

**Issue by nunoperalta** [kamax-matrix/mxisd#126]
I'm using Riot, hosted on my server.

I add an email to my account through Riot.

mxisd sends an email to that address.

I click on the link, and I get a page saying:

"Verification successful: return to your Matrix client to complete the process."
(URL is on my server)

I come back to Riot, refresh page, etc... and email doesn't seem to be added there or verified.

I can send as many emails as I want. It's like nothing is confirming when I click the link from email received.

Who's to blame, though? mxisd or Riot? :)

**Comment by maxidorius** [comment]
It's very possibly a regressing, as we changed a lot in that area for v1.3.0 (the last release). I'll check and come back to you in a  day or so.

**Comment by maxidorius** [comment]
@nunoperalta I've doubled checked and all is fine on mxisd side it seems.

This is what I'm doing to test on Desktop Riot v0.17.9 on Linux Mint 18:
1. Go into settings
2. Type the email I want to add in the empty Email field, press enter
3. Riot tells me to check my emails and then come back when validation is done
4. I receive an email from mxisd with a link
5. I click on the link and end up on a page telling me that verification was successful and return to your Matrix client to complete the process.
6. I go back to Riot, click on the "Continue button"
7. My email is added - Finished.

Can you confirm this is what you are doing also?

**Comment by nunoperalta** [comment]
Aaaahhh!! I didn't realize I had to press "Continue". I didn't even see that button!! I was just refreshing the page to see if the email was added to a list or not.
I'd say that that "Continue" button is an extra step that is not necessary. Once it's confirmed by pressing the link on email, it should be the end of the process :)
That is how it is in all other websites.

Thank you very much for having a look for me. I was able to complete the process now.
